perfkitbenchmarker/benchmarks/tomcat_wrk_benchmark.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run wrk against a simple Tomcat web server.
+
+This is close to HTTP-RR:
+
+  * Connections are reused.
+  * The server does very little work.
+
+Doubles connections up to a fixed count, reports single connection latency and
+maximum error-free throughput.
+
+`wrk` is a scalable web load generator.
+`tomcat` is a popular Java web server.
+"""
+
+import functools
+import logging
+import operator
+import urlparse
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.packages import tomcat
+from perfkitbenchmarker.packages import wrk
+
+
+flags.DEFINE_integer('tomcat_wrk_test_length', 120,
+                     'Length of time, in seconds, to run wrk for each '
+                     'connction count', lower_bound=1)
+flags.DEFINE_integer('tomcat_wrk_max_connections', 128,
+                     'Maximum number of simultaneous connections to attempt',
+                     lower_bound=1)
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'tomcat_wrk'
+BENCHMARK_CONFIG = """
+tomcat_wrk:
+  description: Run wrk against tomcat.
+  vm_groups:
+    server:
+      vm_spec: *default_single_core
+    client:
+      vm_spec: *default_single_core
+"""
+
+# Stop when >= 1% of requests have errors
+MAX_ERROR_RATE = 0.01
```

Comment:
I just retried with/without this, and got more consistent results allowing up to 1% errors. Re-added the error-rate check.